### DISCUSSION TITLE
How to override by new state object

### DISCUSSION THREAD

**Discussion by 1natsu172** [pmndrs/valtio#153] [Q&A] [answered]
I want to re-initialize an existing proxy-state. Is there any way to do this? Or do you have a plan?

![image](https://user-images.githubusercontent.com/7282145/117583720-c3c00a80-b143-11eb-97f6-6bbc09a428d6.png)

Best regards.

**Comment by dai-shi** [comment, marked as answer]
This is not possible as it's how JS (Proxy) works.

Please create a wrapper object.
```js
const  s = proxy({ obj: { hoge: 1, puyo: 2 } })
s.obj = { foo: 1 }
snapshot(s.obj)
```

**Reply by 1natsu172** [reply]
Thank you for your answer.

I see, the only way is to nest the objects. I will try to achieve my goal by changing the State design.
